### ISSUE TITLE
Using the $ accessor on a struct defined in a function with the same name as a script will always give undefined

### ISSUE THREAD

**Issue by KormexGit** [YoYoGames/GameMaker-Bugs#4886]
### Description
This is directly related to #4885. If you define a global struct with the same name as a script in a function and then call that function, using the $ accessor on it will always give undefined, even though struct_get and the dot accessor still work just fine.

### Steps To Reproduce
Put the following in a script, while having a script called Script1:
```
function testFunc(){
	global.Script1 = { 
		hello : 5,
	}
}

testFunc(); 

show_debug_message("struct with struct_get:" + string(struct_get(global.Script1, "hello"))); // shows 5
show_debug_message("struct with $ accessor:" + string(global.Script1[$ "hello"])); //shows undefined
```
The code does not have to be in Script1 specifically, it can be in any script as long as you have a script with the same name.

### Which version of GameMaker are you reporting this issue for?
IDE v2024.2.0.132 Runtime v2024.2.0.163

### Which operating system(s) are you seeing the problem on?
Windows 10.0.22621.0

[9187f381-70ad-4122-857b-5f86ed61aec2](https://api.gamemaker.io/api/github/downloads/9187f381-70ad-4122-857b-5f86ed61aec2)

**Comment by Rhemery** [comment]
To add to this, in my opinion, the issue might be elsewhere. If you have normal struct with member Script1 its fine, but when its on global its not fine anymore? And the fact it worked fine in previous versions.
I mean functions, assets and all live in global scope, but in my opinion "global" and global scope should be different because I (and probably others also) see "global" as a container or just another struct that lives in that global scope.
But it also depends what YoYo were going for in this update. For me personaly, since in IDE that global variable is colored as global variable, it creates some inconsistency.
I think both solution are good and just wanted to give more perspective.

**Comment by rwkay** [comment]
Fixed in 2024.4

**Comment by cameron-home** [comment]
@rwkay What is the fix that was implemented?
